api/src/main/java/marquez/service/models/SourceMeta.java

```diff
@@ -14,18 +14,50 @@
 
 package marquez.service.models;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
 import java.net.URI;
 import java.util.Optional;
 import javax.annotation.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
 import lombok.NonNull;
-import lombok.Value;
+import lombok.ToString;
+import marquez.common.Utils;
 import marquez.common.models.SourceType;
 
-@Value
-public class SourceMeta {
-  @NonNull SourceType type;
-  @NonNull URI connectionUrl;
-  @Nullable String description;
+@EqualsAndHashCode
+@ToString
+public final class SourceMeta {
```

Comment:
Why is this class final? It seems like we would want people to be able to extend the meta class so it can meet their usecase.